examples/units/basic_units.py

```diff
@@ -345,7 +348,20 @@ def convert(val, unit, axis):
         if units.ConversionInterface.is_numlike(val):
             return val
         if np.iterable(val):
-            return [thisval.convert_to(unit).get_value() for thisval in val]
+            if isinstance(val, np.ma.MaskedArray):
+                val = val.astype(float).filled(np.nan)
+            out = np.empty(len(val))
+            for i, thisval in enumerate(val):
+                if np.ma.is_masked(thisval):
```

Comment:
Can values still be masked even though you did `val = val.(etc).filled(np.nan)` if val is a masked array just above?

Reply:
The point here is that convert needs to be able to work on a masked scalar if that is what was provided as the input argument.  If it gets a masked scalar with mask True it needs to replace it with np.nan.

Reply:
This remains not totally clear to me, but in any case it's at worse a dead branch (and much more likely it's just me not understanding masked arrays well enough), so I'm not going to hold up the PR on that.